server/service/src/programs/patient/search_central.rs

```diff
@@ -0,0 +1,118 @@
+use repository::RepositoryError;
+use reqwest::ClientBuilder;
+use url::Url;
+
+use crate::{
+    apis::patient_v4::{
+        NameStoreJoinParamsV4, NameStoreJoinV2, PatientApiV4, PatientParamsV4, PatientV4,
+    },
+    sync::settings::SyncSettings,
+};
+
+use super::PatientSearch;
+
+#[derive(Debug)]
+pub enum CentralPatientRequestError {
+    DatabaseError(RepositoryError),
+    InternalError(String),
+    ConnectionError(String),
+}
+
+pub async fn patient_search_central(
+    sync_settings: &SyncSettings,
+    params: PatientSearch,
+) -> Result<Vec<PatientV4>, CentralPatientRequestError> {
+    let central_server_url = Url::parse(&sync_settings.url).map_err(|err| {
```

Comment:
The reason I did this is that the Client holds a connection pool:

```rust
...
/// The `Client` holds a connection pool internally, so it is advised that
/// you create one and **reuse** it.
...
```

So if you already have a client you can reuse it. Thinking of it, maybe we should actually cache a client? e.g. to avoid expensive ssl handshakes for every request...

Reply:
I thought it could be quite useful, but I am just worried about:
* The effect of poor connectivity of countries we work in, can it cause some issues with connections on connection pool dropping 
* Handling the cache of a client might be a bit more complex, especially since we can change url of the server

Reply:
> So if you already have a client you can reuse it

that makes sense though, i think

Reply:
Yes I can only speculate if the connection pool would help with poor connectivity. Would hope it does something reasonable though. (For example, think connections are usually dropped after some minutes without use...)